src/components/List.js

```diff
@@ -19,6 +19,51 @@ You cannot undo this action, and this item's purchase history will be lost.`,
     return response ? deleteItem(token, result.id) : null;
   }
 
+  function sortedResults() {
+    let soon = [];
+    let kindOfSoon = [];
+    let notSoon = [];
+    let inactive = [];
+
+    results.forEach(result => {
+      //separates each item into arrays by frequency
+      switch (true) {
+        case result.frequency <= 604800000: //milliseconds in 7 days
+          result.timeClass = 'soon';
+          soon.push(result);
+          break;
+        case result.frequency <= 1209600000: //milliseconds in 14 days
+          result.timeClass = 'kind-of-soon';
+          kindOfSoon.push(result);
+          break;
+        case result.frequency <= 2592000000: //milliseconds in 30 days
+          result.timeClass = 'not-soon';
+          notSoon.push(result);
+          break;
+        default:
+          result.timeClass = 'inactive'; //everything above else defaults to inactive
+          inactive.push(result);
+          break;
+      }
```

Comment:
Hi Kat!  We were actually very unsure of how to handle this.  We know that it requested sorting by upcoming purchase date, but what happens when the item is past due?  How long should it be allowed to sit there?  The other really confusing thing is when these items are added, we put them in the categories of soon, kind-of-soon, and not-soon, which reflects their frequency.  Then, we're asked to sort them into groups like that again, but it's not necessarily reflecting the frequency - so an item added as "not soon" might show up as "soon" if it's coming up in 3 days, and that might be confusing to someone looking at labels and thinking they put that as not soon.  Finally, it says that items should be sorted alphabetically only if their days are the exact same, but it probably looks better to do alphabetically in one category.  If we do it just if they have the same date, I don't think it will look organized.  It might look like "  carrots, kiwi, apple, cheese" if carrots and kiwis are coming up sooner than the others.  

This was a little stream-of-consciousness, but please let us know if you have specific ideas because it was very confusing how we should handle that part!

Reply:
> We know that it requested sorting by upcoming purchase date, but what happens when the item is past due? How long should it be allowed to sit there?

I think this is handled by having it show as `soon` until 2 * frequency past its last purchase date, and then it moves to inactive: 

> - Inactive (when there’s only 1 purchase in the database or the purchase is really out of date [the time that has elapsed since the last purchase is 2x what was estimated])

> When these items are added, we put them in the categories of soon, kind-of-soon, and not-soon, which reflects their frequency. Then, we're asked to sort them into groups like that again, but it's not necessarily reflecting the frequency

I think this initial frequency is relative to the date added, not necessarily indicative of a user's perceived purchase frequency, and could be used in tandem with a `dateAdded` value (like what we created in our AC in order to show next purchase date in the details). I think this also speaks to the move to `inactive` once we have one purchase in the purchase history as per the AC (because we don't have enough data to calculate a frequency).

> so an item added as "not soon" might show up as "soon" if it's coming up in 3 days, and that might be confusing to someone looking at labels and thinking they put that as not soon. 

My interpretation was that items added as `not soon` will move through a life cycle of `not soon` -> `kind of soon` -> `soon` -> and then back to `not soon` once the item is purchased or `inactive` if the item is not purchased within the 2 * frequency as per the AC.

> Finally,` it says that items should be sorted alphabetically only if their days are the exact same, but it probably looks better to do alphabetically in one category.

I think that having the items sorted by next purchase date makes more sense from a user perspective when determining how soon an item should be purchased.